### PR TITLE
Clean up tile element usage, introduce struct methods

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

File: src/openrct2/world/TileElement.h [hunk 1]
```diff
@@ -193,22 +195,18 @@ enum
...
-    MAP_ELEM_SMALL_SCENERY_COLOUR_FLAG_NEEDS_SUPPORTS = (1 << 5),
-};
 
-#define TILE_ELEMENT_QUADRANT_MASK 0xC0
-#define TILE_ELEMENT_TYPE_MASK 0x3C
-#define TILE_ELEMENT_DIRECTION_MASK 0x03
+
+#define TILE_ELEMENT_QUADRANT_MASK 0xC0 // bits 6 + 7
+#define TILE_ELEMENT_TYPE_MASK 0x3C     // bits 2 + 3 + 4 + 5
+#define TILE_ELEMENT_DIRECTION_MASK 0x03 // bits 0 + 1
```
Comment (marijnvdwerf) [inline src/openrct2/world/TileElement.h]: Why not use a binary literal instead?

Comment (Gymnasiast) [reply src/openrct2/world/TileElement.h]: That would be fine as well. I do think that if we go down that route, we should put it in the code style rules and adopt this for the whole codebase, for consistency reasons.

**Comment by IntelOrca** [comment]
Most of the C++ written code base uses title case for class / struct methods. Is there a reason you are using lowerCamel?

**Comment by IntelOrca** [comment]
Other than that, it looks fine.

File: src/openrct2/ride/RideRatings.cpp [hunk 2]
```diff
@@ -446,8 +446,8 @@ static void ride_ratings_score_close_proximity_loops_helper(rct_tile_element *in
 
         case TILE_ELEMENT_TYPE_TRACK:
         {
-            sint32 unk = (tileElement->type ^ inputTileElement->type) & 1;
-            if (unk != 0)
+            sint32 elementsAreAt90DegAngle = (tileElement->GetDirection() ^ inputTileElement->GetDirection()) & 1;
```
Comment (marijnvdwerf) [inline src/openrct2/ride/RideRatings.cpp]: bool?

File: src/openrct2/world/TileElement.cpp [hunk 3]
```diff
@@ -23,29 +23,64 @@
...
-uint8 tile_element_get_scenery_quadrant(const rct_tile_element * element)
+uint8 rct_tile_element::GetType() const
 {
-    return (element->type & TILE_ELEMENT_QUADRANT_MASK) >> 6;
+    return this->type & TILE_ELEMENT_TYPE_MASK;
+}
+
+void rct_tile_element::SetType(uint8 newType)
+{
+    this->type |= (newType & TILE_ELEMENT_TYPE_MASK);
```
Comment (duncanspumpkin) [inline src/openrct2/world/TileElement.cpp]: This won't work. You need to clear the type mask first.

File: src/openrct2-ui/windows/TileInspector.cpp [hunk 4]
```diff
@@ -1545,10 +1545,10 @@ static void window_tile_inspector_invalidate(rct_window *w)
...
         w->widgets[WIDX_SCENERY_CHECK_QUARTER_W].bottom = w->widgets[WIDX_SCENERY_CHECK_QUARTER_W].top + 13;
         // This gets the relative rotation, by subtracting the camera's rotation, and wrapping it between 0-3 inclusive
-        bool N = (tileElement->type & TILE_ELEMENT_QUADRANT_MASK) == ((0 - get_current_rotation()) & 3) << 6;
-        bool E = (tileElement->type & TILE_ELEMENT_QUADRANT_MASK) == ((1 - get_current_rotation()) & 3) << 6;
-        bool S = (tileElement->type & TILE_ELEMENT_QUADRANT_MASK) == ((2 - get_current_rotation()) & 3) << 6;
-        bool W = (tileElement->type & TILE_ELEMENT_QUADRANT_MASK) == ((3 - get_current_rotation()) & 3) << 6;
+        bool N = tileElement->GetSceneryQuadrant() == ((0 - get_current_rotation()) & 3);
+        bool E = tileElement->GetSceneryQuadrant() == ((1 - get_current_rotation()) & 3);
+        bool S = tileElement->GetSceneryQuadrant() == ((2 - get_current_rotation()) & 3);
+        bool W = tileElement->GetSceneryQuadrant() == ((3 - get_current_rotation()) & 3);
```
Comment (Broxzier) [inline src/openrct2-ui/windows/TileInspector.cpp]: The outer parentheses are superfluous now.

Comment (Gymnasiast) [reply src/openrct2-ui/windows/TileInspector.cpp]: I think it's clearer this way.

**Pull request merged**